### PR TITLE
KEYCLOAK-19086 Default Client Roles are not implicitly created

### PR DESCRIPTION
**Pull request by vmuzikar**
This PR introduces a workaround for adding default clients role. Keycloak REST API no longer respects the `defaultRoles` field in Clients. It is not a problem as such as there's another way to add default roles through the REST API (adding a composite roles to certain default realm role that acts as an umbrella for all default roles). However, it was not possible to do this through the Operator. This PR introduces a way to "mimic" the old behaviour of `defaultRoles` field in Clients.

See [KEYCLOAK-19086](https://issues.redhat.com/browse/KEYCLOAK-19086) for more details.

### CODE REVIEW

File: pkg/common/client.go [hunk 1]
```diff
@@ -98,6 +98,11 @@ func (c *Client) CreateClientRole(clientID string, role *v1alpha1.RoleRepresenta
 	return c.create(role, fmt.Sprintf("realms/%s/clients/%s/roles", realmName, clientID), "client role")
 }
 
+func (c *Client) AddRealmRoleComposites(realmName, roleID string, roles *[]v1alpha1.RoleRepresentation) error {
```
[the remark this answers is not in the document]
Comment (vmuzikar) [reply pkg/common/client.go:101]: I'd say not. We're not creating any new resource, we're basically just adding a link between already existing resources. The same name scheme is used in KC as well, compare [`createRole`](https://github.com/keycloak/keycloak/blob/b1bcd5d66e148eee4331e2a2f60a8080761ff8ee/services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java#L131) and [`addComposites`](https://github.com/keycloak/keycloak/blob/b1bcd5d66e148eee4331e2a2f60a8080761ff8ee/services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java#L297). Let me know if that makes sense.

File: pkg/common/cluster_actions.go [hunk 2]
```diff
@@ -322,6 +324,20 @@ func (i *ClusterActionRunner) RemoveClientRole(obj *v1alpha1.KeycloakUserRole, c
 	return i.keycloakClient.DeleteUserClientRole(obj, realm, clientID, userID)
 }
 
+func (i *ClusterActionRunner) AddDefaultRoles(obj *[]v1alpha1.RoleRepresentation, defaultRealmRoleID, realm string) error {
```
[the remark this answers is not in the document]
Comment (vmuzikar) [reply pkg/common/cluster_actions.go:327]: I think "create" would imply we're really creating the roles while we are not. We're just setting existing roles as default, technically just adding a composite to the "umbrella" default role in realm. Does it make sense?

File: pkg/common/cluster_actions.go [hunk 3]
```diff
@@ -444,6 +460,22 @@ type DeleteClientRoleAction struct {
 	Realm string
 }
 
+type AddDefaultRolesAction struct {
```
[the remark this answers is not in the document]
Comment (vmuzikar) [reply pkg/common/cluster_actions.go:463]: I think "create" would imply we're really creating the roles while we are not. We're just setting existing roles as default, technically just adding a composite to the "umbrella" default role in realm. Does it make sense?

**Comment by vmuzikar** [comment]
@jonathanvila Thanks for the review!

**Review by stianst — APPROVED** [review]

**Pull request merged**